src/main/java/com/example/popcorn_api/controllers/CategoriesController.java

```diff
@@ -42,26 +44,39 @@ public ResponseEntity<List<Categories>> getAllCategory() {
     private ResponseEntity<?>getCategory(@PathVariable Long id) {
         try{
             return  ResponseEntity.ok(categoriesService.findCategoryById(id));
-        }catch(CategoryNotFoundException e){
+        }catch(CategoryException e){
             return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
         }
        
     }
     @PostMapping
-    public void createCategory(@RequestBody CategoryDTO  categoryDTO) {
+    public ResponseEntity<?> createCategory(@RequestBody @Valid CategoryDTO  categoryDTO) {
         Categories category = new Categories();
-        category.setName(categoryDTO.getName());
-        categoriesService.create(category);
         
+        category.setName(categoryDTO.getName());
+        try{
+        return ResponseEntity.status(HttpStatus.CREATED).body(categoriesService.create(category));
```

Comment:
This should be written this way: 
Categories categorie = categoriesService.create(category);
return ResponseEntity.status(HttpStatus.CREATED).body(categorie);

I think it's better.

Reply:
I've updated for updatedCategory and pass throught ResponseEntity body. Check it out ;)